app/src/main/java/com/movies/presentation/home/view_model/HomeViewModel.kt

```diff
@@ -23,53 +28,68 @@ class HomeViewModel(
     private val getFavouriteMovies: GetFavouriteMoviesUseCase,
 ) : BaseViewModel() {
 
-    val fetchMoviesLiveData by LiveDataDelegate<List<MovieUIModel>>()
     val loadingLiveData by LiveDataDelegate<Boolean>()
-    val searchMoviesLiveData by LiveDataDelegate<List<MovieUIModel>>()
     val fetchFavouriteMoviesLivedata by LiveDataDelegate<List<MovieUIModel>>()
-    private val categoryStateLiveData = MutableLiveData(CategoryType.POPULAR)
+
+    private val _fetchMoviesStateFlow = MutableStateFlow<PagingData<MovieUIModel>?>(null)
+    val fetchMoviesStateFlow get() = _fetchMoviesStateFlow.asStateFlow()
+
+    private val _categoryStateFlow = MutableStateFlow(CategoryType.POPULAR)
+    private val categoryStateFlow = _categoryStateFlow.asStateFlow()
+
+    private val _searchStateFlow = MutableStateFlow<PagingData<MovieUIModel>?>(null)
+    val searchStateFlow = _searchStateFlow.asStateFlow()
```

Comment:
Can you use one StateFlow for fetch movies and searching movies?

Reply:
did in network PR